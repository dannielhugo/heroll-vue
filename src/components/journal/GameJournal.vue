<template>
  <div v-loading="loading" class="journal-container">
    <GridView :games="games" @detail="onDetail" />
  </div>
</template>

<script setup lang="ts">
import type { Game } from '@/models/game';
import GridView from './grid-view/GridView.vue';
defineProps<{ games: Game[]; loading: boolean }>();

const emit = defineEmits(['detail']);

const onDetail = (game: Game) => {
  emit('detail', game);
};
</script>

<style scoped lang="scss">
.journal-container {
  width: 100%;
}
</style>