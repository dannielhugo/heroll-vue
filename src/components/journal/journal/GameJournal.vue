<template>
  <div v-loading="loading" class="container">
    <GridView :games="games" />
  </div>
</template>

<script setup lang="ts">
import { watch, ref } from 'vue';
import { storeToRefs } from 'pinia';
import { useCatalogStore } from '@/stores/catalog';
import { useUserStore } from '@/stores/user';
import type { User } from '@/models/user';
import type { Game } from '@/models/game';
import GridView from '../grid-view/GridView.vue';
import type { Catalog } from '@/models/catalog';

const userStore = useUserStore();
const catalogStore = useCatalogStore();
const { user } = storeToRefs(userStore);
const { loading, catalog } = storeToRefs(catalogStore);
const { load } = catalogStore;
const games = ref<Game[]>(catalog?.value?.games || []);

watch(user, async (newValue: User) => {
  if (newValue) {
    await load(newValue.id);
  }
});

watch(catalog, (newValue: Catalog) => {
  games.value = newValue.games;
});
</script>

<style scoped lang="scss">
.container {
  height: 100%;
  width: 100%;
  margin: auto;

  @media (min-width: 1536px) {
    max-width: 1536px;
    padding-left: 24px;
    padding-right: 24px;
  }
}
</style>
