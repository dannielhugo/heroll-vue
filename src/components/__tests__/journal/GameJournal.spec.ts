import { mount } from '@vue/test-utils';
import ElementPlus from 'element-plus';
import GameJournal from '../../journal/GameJournal.vue';

describe('GameJournal', () => {
  test('should mount component', async () => {
    expect(GameJournal).toBeTruthy();

    const wrapper = mount(GameJournal, {
      props: {
        loading: false,
        games: [],
      },
      global: {
        plugins: [ElementPlus],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
