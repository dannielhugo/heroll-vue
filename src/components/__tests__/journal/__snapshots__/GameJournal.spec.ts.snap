// Vitest Snapshot v1

exports[`GameJournal > should mount component 1`] = `
"<div class=\\"container\\" data-v-06c35f8d=\\"\\">
  <div class=\\"game-list\\" data-v-ef199a12=\\"\\" data-v-06c35f8d=\\"\\"></div>
</div>"
`;
